lib/Onyx.js

```diff
@@ -374,7 +375,7 @@ function keyChanged(key, data) {
     const stateMappingKeys = _.keys(callbackToStateMapping);
     for (let i = stateMappingKeys.length; i--;) {
         const subscriber = callbackToStateMapping[stateMappingKeys[i]];
-        if (!subscriber || !isKeyMatch(subscriber.key, key)) {
+        if (!subscriber || !isKeyMatch(subscriber.key, key) || (_.isFunction(canUpdateSubscriber) && !canUpdateSubscriber(subscriber))) {
```

Comment:
NAB: Is this new logic the same thing as using `_.result(subscriber, 'canUpdateSubscriber')`? https://underscorejs.org/#result

Maybe not... I think I am thinking of this: https://github.com/Expensify/expensify-common/blob/main/lib/Func.jsx#L13-L19 which would be cool if we could just import it without any additional work :D I don't think that's possible though.

Reply:
[`_.result()` takes a property name as it's second arg](https://underscorejs.org/#result) so it's a little different

What you are asking sounds similar to [`Str.result()`](https://github.com/Expensify/expensify-common/blob/e550f5940461edb08b0b3edf6c2f8d64ad33442c/lib/str.js#L994-L1006) and I think would work if we did:

`Str.result(canUpdateSubscriber, subscriber)`

and would return `undefined` if the first argument was `undefined`